lib/granite/base.rb

```diff
@@ -0,0 +1,37 @@
+require 'active_data/model'
+require 'active_data/model/primary'
+require 'active_data/model/lifecycle'
+require 'active_data/model/associations'
+
+require 'granite/translations'
+require 'granite/represents'
+
+module Granite
+  module Base
+    extend ActiveSupport::Concern
+
+    include ActiveSupport::Callbacks
+    include ActiveData::Model
+    include ActiveData::Model::Representation
+    include ActiveData::Model::Dirty
+    include ActiveData::Model::Associations
+    include ActiveData::Model::Primary
+    include ActiveModel::Validations::Callbacks
+
+    include Granite::Translations
+    include Granite::Represents
+
+    included do
+      def self.i18n_scope
+        :granite_action
+      end
+    end
+  end
+
+  module BaseConcern
+    extend ActiveSupport::Concern
+
+    include ActiveData::Model::Lifecycle
+    include Base
+  end
+end
```

Comment:
Oo why do we have two of them again?

Reply:
because `ActiveData::Model::Lifecycle` is required for `embeds_many`, but not for `Action`

Reply:
if we put `ActiveData::Model::Lifecycle` to the `Action` we will have:
```ruby
     ActiveData::UnsavableObject:
       ActiveData::UnsavableObject
     # /home/bioform/.rvm/gems/ruby-2.4.3/bundler/gems/active_data-8b0df4d47763/lib/active_data/model/lifecycle.rb:198:in `save'
     # /home/bioform/.rvm/gems/ruby-2.4.3/bundler/gems/active_data-8b0df4d47763/lib/active_data/model/lifecycle.rb:160:in `update'
     # ./spec/lib/granite/action_spec.rb:167:in `block (5 levels) in <top (required)>'
     # ./spec/lib/granite/action_spec.rb:167:in `block (4 levels) in <top (required)>'
```